CMakeLists.txt

```diff
@@ -230,32 +230,74 @@ IF(NOT CYCLUS_DOC_ONLY)
     #  COMPONENT data
     # )
 
-    # Here we set some components for installation with cpack
-    SET(CPACK_COMPONENTS_ALL cycamore testing libraries data core)
 
+    ##############################################################################################
+    ################################## begin uninstall target ####################################
+    ##############################################################################################
+ 
+    CONFIGURE_FILE(
+        "${CMAKE_CURRENT_SOURCE_DIR}/config/cmake_uninstall.cmake.in"
+        "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
+        IMMEDIATE @ONLY
+        )
+
+    ADD_CUSTOM_TARGET(uninstall
+        COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
+
+    ##############################################################################################
+    ################################### end uninstall target #####################################
+    ##############################################################################################
+ 
+    ##############################################################################################
+    ####################################### begin cpack ##########################################
+    ##############################################################################################
+    
+    
     # Here's where we package it with CPack
     SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Cycamore : The Cyclus Additional Module Repository.")
+    
+    # Here we set some components for installation with cpack
+    SET(CPACK_COMPONENTS_ALL cycamore testing libraries data core)
+    SET(CPACK_GENERATOR "DEB")
+
+    # Organisation
     SET(CPACK_DEBIAN_PACKAGE_MAINTAINER "CNERG, UW-Madison") #required
     SET(CPACK_PACKAGE_VENDOR "CNERG, UW-Madison")
     SET(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README.rst")
     SET(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.rst")
+    
+    # Version
     SET(CPACK_PACKAGE_VERSION_MAJOR "1")
-    SET(CPACK_PACKAGE_VERSION_MINOR "0")
-    #SET(CPACK_PACKAGE_VERSION_PATCH "0") # should use commit number here  ?
-    SET(CPACK_PACKAGE_INSTALL_DIRECTORY "cycamore${CPACK_VERSION_MAJOR}.${CPACK_VERSION_MINOR}")
-    SET(CPACK_PACKAGE_NAME "${CMAKE_PROJECT_NAME}")
+    SET(CPACK_PACKAGE_VERSION_MINOR "4")
+    SET(CPACK_PACKAGE_VERSION_PATCH "0")
+    
+    # Names
+    SET(CPACK_PACKAGE_NAME "cycamore")
+    SET(
+      CPACK_PACKAGE_INSTALL_DIRECTORY
+      "cycamore_${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}"
+      )
+    
+    SET(
+      CPACK_PACKAGE_FILE_NAME
+      "${CPACK_PACKAGE_NAME}_${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}_${CPACK_HASH_NAME}_${CPACK_SYSTEM_NAME}"
```

Comment:
Similar long line question to my comment in https://github.com/cyclus/cyclus/pull/1259